cmd/containerd-shim-runhcs-v1/exec_hcs.go

```diff
@@ -290,7 +290,22 @@ func (he *hcsExec) Kill(ctx context.Context, signal uint32) error {
 		}
 		var delivered bool
 		if supported && options != nil {
-			delivered, err = he.p.Process.Signal(ctx, options)
+			if he.isWCOW {
+				// We deliver the signal to the particular process in a separate background
```

Comment:
It would be nice to have more information here, such as:
- Why do we need to do this to gracefully terminate containers (because servercore will otherwise block on signaling waiting for the process to terminate)
- The reason this will work (because now the caller can wait for their desired timeout and then send SIGKILL)